### PR TITLE
Improve foss serial commands

### PR DESCRIPTION
**Pull request by iamashwin99**

### CODE REVIEW

**Comment by iamashwin99** [comment]
@fangohr could you please take a look at the use of the docker file and install script and let me know if this is a good idea?
Features/bug of this approach : 
- no obvious bare metal variant
- compile toolchains as an argument to make :
```shell
# mpi toolchain
make docker-build-octopus TOOLCHAIN="foss2022a-mpi"
# and
# serial toolchain
make docker-build-octopus TOOLCHAIN="foss2022a-serial"
# or
make docker-build-octopus

**Comment by fangohr** [comment]
Thanks! This is certainly progress: it runs through; at least for the serial toolchain. 

I suggest we merge this (after addressing the things I spotted above), and brain storm how to proceed from here in person.

**Review by fangohr — APPROVED** [review]
Nice! Interesting idea with the multi-stage dockerfiles.

See comments (and please address) before merging.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+              git \
+              openssh-server \
+    python \
+              unzip
+
+# Convenience tools, if desired for debugging etc
+RUN apt-get -y install wget time nano vim emacs vim
+
+# Tools needed by mpsd-software-environment.py (and ../spack-setup.sh)
+RUN apt-get -y install rsync automake libtool linux-headers-amd64
```
[the remark this answers is not in the document]
Comment (iamashwin99) [reply Dockerfile:31]: I tried to compile the MPI version but couldnt get it working

[the remark this answers is not in the document]
Comment (fangohr) [reply Dockerfile:31]: I would expect that on github (as it is unlikely you have the kernel headers for the machine that is hosting the docker container, and then the kmem compilation will fail).

I am just trying to run this on avocado for the mpi toolchain ...

It should work in the buildbots as well?

**Pull request merged**